src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicDocumentPullDiagnosticsHandler_IOnInitialized.cs

```diff
@@ -2,61 +2,44 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.SolutionCrawler;
 using Roslyn.LanguageServer.Protocol;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.Public;
-
-using DocumentDiagnosticReport = SumType<RelatedFullDocumentDiagnosticReport, RelatedUnchangedDocumentDiagnosticReport>;
-
 // A document diagnostic partial report is defined as having the first literal send = DocumentDiagnosticReport (aka changed / unchanged) followed
 // by n DocumentDiagnosticPartialResult literals.
 // See https://github.com/microsoft/vscode-languageserver-node/blob/main/protocol/src/common/proposed.diagnostics.md#textDocument_diagnostic
-using DocumentDiagnosticPartialReport = SumType<RelatedFullDocumentDiagnosticReport, RelatedUnchangedDocumentDiagnosticReport, DocumentDiagnosticReportPartialResult>;
 
-internal sealed partial class PublicDocumentPullDiagnosticsHandler : AbstractDocumentPullDiagnosticHandler<DocumentDiagnosticParams, DocumentDiagnosticPartialReport, DocumentDiagnosticReport?>, IOnInitialized
+internal sealed partial class PublicDocumentPullDiagnosticsHandler : IOnInitialized
 {
     public async Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
     {
-        // Dynamically register a non-local document diagnostic source if Full solution background analysis is enabled
-        // for analyzer execution. This diagnostic source reports diagnostics in open documents that are reported
-        // when analyzing other documents or at compilation end.
-        if (clientCapabilities?.TextDocument?.Diagnostic?.DynamicRegistration is true && IsFsaEnabled())
+        // Dynamically register for all of our document diagnostic sources.
+        if (clientCapabilities?.TextDocument?.Diagnostic?.DynamicRegistration is true)
         {
             // TODO: Hookup an option changed handler for changes to BackgroundAnalysisScopeOption
             //       to dynamically register/unregister the non-local document diagnostic source.
 
+            var sources = DiagnosticSourceManager.GetSourceNames(isDocument: true).Where(source => source != PullDiagnosticCategories.Task);
```

Comment:
need explaantion on why we're explicitly filtering out Task here.  Part of the idea of this being an opaque extensibility system is that the core parts shouldn't know about the extensions composed into them (unless they interrogate them directly).

Reply:
for example, why are we only filtering out task here?  why not EnC?  or xaml?  etc.

Reply:
task diagnostics shouldn't be reported through VSCode (it has its own task stuff).  Maybe there is a better way to somehow mark some sources as VS only, I would be OK with cleaning that up in a followup myself.

Reply:
can we add an IsVSSpecific property to the interface?  then the public handlers can filter those out.